ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorGroupByOperator.java

```diff
@@ -358,6 +360,34 @@ public void close(boolean aborted) throws HiveException {
      */
     private long numRowsCompareHashAggr;
 
+    /**
+     * To track current memory usage.
+     */
+    private long currMemUsed;
+
+    /**
+     * Whether to make use of LRUCache for map aggr buffers or not.
+     */
+    private boolean lruCache;
+
+    class LRUCache extends LinkedHashMap<KeyWrapper, VectorAggregationBufferRow> {
```

Comment:
this doesn't look like an `LRU` cache to me...it removes the eldest entry; in case of `lru` you should move the accessed entry to the head of the list

I think guava has some lru cache implementation - might worth a look

Reply:
LinkedHashMap provides this semantics and already maintains the double linked list internally. Invokes removeEldestEntry on need basis.  However, planning to make it a lot more simpler in next iteration of the patch.

Reply:
okay; but this is not an `LRU` cache; please rename the class/option/etc because it may only cause confusion